src/libraries/Common/tests/System/Security/Cryptography/X509Certificates/CertificateAuthority.cs

```diff
@@ -929,10 +944,14 @@ private enum CertStatus
                     altName = builder.Build();
                 }
 
-                endEntityCert = intermediateAuthority.CreateEndEntity(
-                    BuildSubject(subjectName ?? "A Revocation Test Cert", testName, pkiOptions, pkiOptionsInSubject),
-                    eeKey,
-                    altName);
+                endEntityCert = endEntityIsServer ?
+                    intermediateAuthority.CreateServerEndEntity(
```

Comment:
Really this feels like it should just take an OidCollection or Oid or string, rather than using a bit in the PkiOptions.

Reply:
I was thinking about adding bool to BuildPrivatePki() but OidCollection seems like better option. I think more variations are yet to come.